fieldtracing/fieldtracing.cpp

```diff
@@ -996,114 +1010,213 @@ namespace FieldTracing {
+            #pragma omp for schedule(dynamic) reduction(+:cellsToDoFullBox) reduction(+:cellsToDoFluxRopes)
             for(int n=0; n<globalDccrgSize; n++) {
-               if(reducedCellNeedsContinuedFWTracing[n] > 0) {
-                  anyCellNeedsTracing=true;
-                  cellNeedsContinuedFWTracing[n] = 1;
-                  cellsToDo++;
-                  
+               if(cellNeedsContinuedFWTracing[n] > 0) {
+                  cellsToDoFullBox++;
                   // Update that nodes' tracing coordinates
                   cellFWTracingCoordinates[n][0] = sumCellFWTracingCoordinates[n][0] / reducedCellNeedsContinuedFWTracing[n];
                   cellFWTracingCoordinates[n][1] = sumCellFWTracingCoordinates[n][1] / reducedCellNeedsContinuedFWTracing[n];
                   cellFWTracingCoordinates[n][2] = sumCellFWTracingCoordinates[n][2] / reducedCellNeedsContinuedFWTracing[n];
-                  
-                  cellFWStepCounter[n] = maxCellFWStepCounter[n];
+                  if(cellFWConnection[n] == TracingLineEndType::UNPROCESSED) {
+                     cellsToDoFluxRopes++;
+                  }
                }
-               if(reducedCellNeedsContinuedBWTracing[n] > 0) {
-                  anyCellNeedsTracing=true;
-                  cellNeedsContinuedBWTracing[n] = 1;
-                  cellsToDo++;
-                  
+               if(cellNeedsContinuedBWTracing[n] > 0) {
+                  cellsToDoFullBox++;
                   // Update that nodes' tracing coordinates
                   cellBWTracingCoordinates[n][0] = sumCellBWTracingCoordinates[n][0] / reducedCellNeedsContinuedBWTracing[n];
                   cellBWTracingCoordinates[n][1] = sumCellBWTracingCoordinates[n][1] / reducedCellNeedsContinuedBWTracing[n];
                   cellBWTracingCoordinates[n][2] = sumCellBWTracingCoordinates[n][2] / reducedCellNeedsContinuedBWTracing[n];
-                  
-                  cellBWStepCounter[n] = maxCellBWStepCounter[n];
+                  if(cellBWConnection[n] == TracingLineEndType::UNPROCESSED) {
+                     cellsToDoFluxRopes++;
+                  }
                }
-               cellFWTracingStepSize[n] = reducedCellFWTracingStepSize[n];
-               cellBWTracingStepSize[n] = reducedCellBWTracingStepSize[n];
             }
             #pragma omp barrier
-         } while(anyCellNeedsTracing && (cellsToDo >= fieldTracingParameters.max_incomplete_lines_fullbox * 2 * globalDccrgSize));
+         } while(!(
+            cellsToDoFullBox <= fieldTracingParameters.fullbox_max_incomplete_lines * 2 * globalDccrgSize
+            && cellsToDoFluxRopes <= fieldTracingParameters.fluxrope_max_incomplete_lines * 2 * globalDccrgSize
+         ));
          
          // Last pass to sort cells that would still continue but won't as we exited as TracingLineEndType::LOOP instead of UNPROCESSED.
```

Comment:
This seemed like a good idea, less so now. The larger the leftover fraction we set, the less sensible this is. Conversely with a small fraction we're more likely to have indeed leftover loops mostly. Opinions? Leave unprocessed? Set to loop? Make the choice a cfg option?

Reply:
Hmm, I'm personally in favour of explicitly marking them as UNPROCESSED, just to be able to afterwards quantify how confident we are with the result. In most cases, we'll then afterwards just treat them the same way anyway.

Reply:
OK. I'll wait for others' opinion but I agree it'd be best to leave them tagged as UNPROCESSED.

Reply:
I agree - if it's a small fraction, then we can indeed re-evaluate them in post.

Reply:
OK pushed this, removed this last pass.